From fd02dcdc7a0faaf972f6c9e8eda3f44b24dda337 Mon Sep 17 00:00:00 2001
From: Adrian Reber <adrian@lisas.de>
Date: Fri, 7 Feb 2014 16:11:50 +0100
Subject: [PATCH] Makefile: fix libcriu.so links and man page installation

The links to the versioned libcriu.so.* included $(DESTDIR) and
$(LIBDIR) which will generate wrong links if building criu packages.
Now the links points to the correct file without any path.

For some reason criu.8 was installed with executable permission. Use
'install' with '-m 644' to make sure this does not happen.

Signed-off-by: Adrian Reber <adrian@lisas.de>
---
 Documentation/Makefile | 2 +-
 Makefile               | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Documentation/Makefile b/Documentation/Makefile
index 50e4526..e236635 100644
--- a/Documentation/Makefile
+++ b/Documentation/Makefile
@@ -46,6 +46,6 @@ clean:
 install: $(MANS)
 	$(E) "  INSTALL " $(MANS)
 	$(Q) mkdir -p $(DESTDIR)$(MAN8DIR)
-	$(Q) install $(MANS) $(DESTDIR)$(MAN8DIR)
+	$(Q) install -m 644 $(MANS) $(DESTDIR)$(MAN8DIR)
 
 .PHONY: clean install
diff --git a/Makefile b/Makefile
index 9ed45bd..a4d560a 100644
--- a/Makefile
+++ b/Makefile
@@ -241,9 +241,9 @@ install: $(PROGRAM) install-man
 	$(Q) mkdir -p $(DESTDIR)$(LIBDIR)
 	$(Q) install -m 755 $(CRIU-LIB) \
 		$(DESTDIR)$(LIBDIR)/$(CRIU-SO).so.$(VERSION_SO_MAJOR).$(VERSION_SO_MINOR)
-	$(Q) ln -s $(DESTDIR)$(LIBDIR)/$(CRIU-SO).so.$(VERSION_SO_MAJOR).$(VERSION_SO_MINOR) \
+	$(Q) ln -s $(CRIU-SO).so.$(VERSION_SO_MAJOR).$(VERSION_SO_MINOR) \
 		$(DESTDIR)$(LIBDIR)/$(CRIU-SO).so.$(VERSION_SO_MAJOR)
-	$(Q) ln -s $(DESTDIR)$(LIBDIR)/$(CRIU-SO).so.$(VERSION_SO_MAJOR).$(VERSION_SO_MINOR) \
+	$(Q) ln -s $(CRIU-SO).so.$(VERSION_SO_MAJOR).$(VERSION_SO_MINOR) \
 		$(DESTDIR)$(LIBDIR)/$(CRIU-SO).so
 	$(Q) mkdir -p $(DESTDIR)$(INCLUDEDIR)
 	$(Q) install -m 644 $(CRIU-INC) $(DESTDIR)$(INCLUDEDIR)
-- 
1.8.5.3

